swaps/swaps_service.go

```diff
@@ -699,31 +694,78 @@ func (svc *swapsService) RefundSwap(swapId, address string) error {
 		return err
 	}
 
-	fastestFee := float64(feeRates.FastestFee)
-	refundTransaction, _, err := boltz.ConstructTransaction(
-		network,
-		boltz.CurrencyBtc,
-		[]boltz.OutputDetails{
-			{
-				SwapId:             swapId,
-				SwapType:           boltz.NormalSwap,
-				Address:            address,
-				LockupTransaction:  lockupTransaction,
-				TimeoutBlockHeight: swapTransactionResp.TimeoutBlockHeight,
-				Vout:               vout,
-				PrivateKey:         ourKeys,
-				SwapTree:           tree,
-				Cooperative:        true,
+	waitToRetrySeconds := 0
+	var refundTransaction boltz.Transaction
+
+	for i := 0; ; i++ {
+		contextCancelled := false
+		select {
+		case <-svc.ctx.Done():
+			logger.Logger.WithField("swapId", swapId).Info("Swap refund context cancelled")
+			contextCancelled = true
```

Comment:
why not `return nil` here? is contextCancelled needed?

Reply:
Ah true, changing!